### PR TITLE
Fix Bundle #1: List Component refactored, Locale State System shaved down, Header default fallback, Sidenav link issue. 

### PR DESCRIPTION
**Pull request by MaximumCrash**
### Attention (CC)
@twblack88 
@andytudhope 
@isaacpearl 

## Summary of changes
This is part 1 of the "Big Fix" series of polish and edge case findings as we start to have content creators build content for the site. 

- Refactored and improved the stability of the List component. 
- Refactored the locale statemanagement to trust the path/url provided instead of storing state in the context api. 
- Fixed the Breadcrumbs from crashing due to missing in-between pages edge case. 
- Updated the Language selector for improved stability with new locale changes. 
- Fixed the Sidenav on non default locales causing a complete overlap of the default locale.

## Motivation & Context
- Issue: #54, #56, #74, #75, #76

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation Update
- [ ] Other (please describe):

## Does this introduce a breaking change?

_Breaking change is defined as_ **"A fix or feature that would cause existing functionality to change."**

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added/changed necessary documentation (if appropriate)
- [ ] All new and existing tests passed.

## How Has This Been Tested?
Locally

### List Components and writing in Cursive

The previous iteration of the list component became unstable due to the pickiness of how MDX, and MD in general, handles spaces and newlines to designate differences between changes. There is 1 very specific edge case that I believe is an inconsistency in the spec:

```md
<List>

   - Header text
       - <Box> 
           
        ### More complicated MD ... 
           
        </Box>

</List>
```

**AND**

```md
<List>

   - Header text

       - <Box> 
           
        ### More complicated MD ... 
           
        </Box>

</List>
``` 

would give us different results. This is because there are times where (depending on the spacing or tabs) MD will recognize the <Box> component as a **sibling** or as a **child** of the list. Due to this, I've had the list component's logic for checking for links and sibilings updated to support the case where we have this inconsistency. 

TLDR: Sibling list elements at the same top level as the list header will now be treated as individual list elements as we would expect them to be. 

### Troubles with Locales

**TLDR: Trust the path, don't interweave other locales into others it's just not cool mang.** 

Multiple components, specifically the Language Selector, Sidenav, and Breadcrumbs component (all dependent on locale), were showing inconsistent results in different locales due to how we were handling the initial state for locales. 

Originally in the PageContext file we would use "useState" for our locales with 2 useLayoutEffects that checked if our localStorage locale matched our current locale and if our current url route had a different locale than our current (if it did then update the state). There 2 problems with this line of thinking, in regards to this kind of app:

1. Every time a person went to a new page the locales would jump "EN" (initial state and default locale) -> "EN" -> "FR" (current true locale)

2. If a person who starts the site in German goes to an english fallback page, their state locale in local storage is updated to English instead of staying in their own locale. (<-- This is a somewhat in-escapable situation)

Too many re-renders and the inconsistency of the true current locale ment that pure components like Sidenav (that means they render based on shallow comparisons of their props and state) would render in the default language **and stay that way**. Or in the case of breadcrumbs or language selector you'd see inconsistencies in label rendering and available languages. 

The issue with our #2 problem is that due to the fallback nature of our application (because a graceful fallback to EN is way better than crashing our site) we can't support interwoven/mixed content in different locales. It is not possible to have a route thats **/fr/work_with_us/governance?locale=en** or **/fr/work_with_us/en/governance**. Our philosophy is what you see is what you get and unless we plan for content creators also manage interwoven content, this is a hard ask (and also not the most elegant of solutions). 

From here on out our locale will forever be strictly route based. We grab the locale from the pathname and provide it through our useTranslation hook. I'm slowly shaving down PageContext to where we can do away with contextual state for this application. 

### Header Fallbacks

In the unique case where a current locale has **NO** pages with ```header: true``` in their frontmatter the header now falls back to the English header generation. The current locale needs **at least** one page with ```header: true``` to render it's own header. 

### Sidenav woes

The sidenav experienced some inconsistency due to the locale statefulness for pure/functional components. While this has been allieviated there is another issue I noticed while fixing it. I've logged it here: #82

### CODE REVIEW

File: src/modules/utility/Video.js [hunk 1]
```diff
@@ -5,10 +5,10 @@ import {Box, jsx} from 'theme-ui'
 //NOTE(Rejon): This is to support playlists for youtube embed specifically!
 //This video component is a basic recreation of what the remark-gatsby-video plugin creates. 
 //We inline the allowances and styles as needed, but this can be used for anything that can be embed into an iframe. 
-const Video = ({src}) => (
+const Video = ({src, title}) => (
```
Comment (twblack88) [inline src/modules/utility/Video.js:8]: 😄

**Review by twblack88 — APPROVED** [review]
Looks like a lot of cleanup!
